ext/date/php_date.c

```diff
@@ -4290,34 +4291,44 @@ PHP_FUNCTION(timezone_transitions_get)
 		add_assoc_long(&element, "offset", tzobj->tzi.tz->type[0].offset); \
 		add_assoc_bool(&element, "isdst",  tzobj->tzi.tz->type[0].isdst); \
 		add_assoc_string(&element, "abbr", &tzobj->tzi.tz->timezone_abbr[tzobj->tzi.tz->type[0].abbr_idx]); \
-		add_next_index_zval(return_value, &element);
+		add_next_index_zval(return_value, &element); \
+		timestamp_added_last = timestamp_begin;
 
 #define add(i,ts) \
-		array_init(&element); \
-		add_assoc_long(&element, "ts",     ts); \
-		add_assoc_str(&element, "time", php_format_date(DATE_FORMAT_ISO8601_LARGE_YEAR, 13, ts, 0)); \
-		add_assoc_long(&element, "offset", tzobj->tzi.tz->type[tzobj->tzi.tz->trans_idx[i]].offset); \
-		add_assoc_bool(&element, "isdst",  tzobj->tzi.tz->type[tzobj->tzi.tz->trans_idx[i]].isdst); \
-		add_assoc_string(&element, "abbr", &tzobj->tzi.tz->timezone_abbr[tzobj->tzi.tz->type[tzobj->tzi.tz->trans_idx[i]].abbr_idx]); \
-		add_next_index_zval(return_value, &element);
+		if (timestamp_added_last != ts) { \
```

Comment:
Wouldn't it make more apparent why this is done, by using:

```
if (ts > timestamp_added_last) {  \
```

They should always be sequential.

Reply:
while I don't see where it makes it in any way more apparent I also don't see an issue with it.

Reply:
updated